### PR TITLE
use base64 crate for encoding

### PR DESCRIPTION
**Pull request by danieleades**
use the `base64` crate, rather than hand-rolling this

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -40,6 +40,7 @@ deflate = { version = "0.9.1", optional = true }
 sha1 = { version = "0.6.0", features = ["std"] }
 tempfile = "3.2.0"
 futures = "0.3.17"
+base64 = "=0.20.0-alpha.1"
```
Comment (danieleades) [inline Cargo.toml:43]: this is an alpha release, though the author suggests that the API and implementation are stable. I've pinned the version to mitigate any risk here.

see https://github.com/marshallpierce/rust-base64/issues/174#issuecomment-1032751831

File: src/base64_plantuml.rs [hunk 2]
```diff
@@ -1,68 +1,19 @@
...
+use base64::{
+    alphabet::Alphabet,
+    engine::fast_portable::{self, FastPortable},
+};
+
+const ENGINE: FastPortable =
+    match Alphabet::from_str("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-_") {
+        Ok(alphabet) => FastPortable::from(&alphabet, fast_portable::PAD),
+        Err(_e) => unreachable!(),
+    };
```
Comment (danieleades) [inline src/base64_plantuml.rs:12]: this construction is needed since `Result::unwrap` is not (yet) `const`

**Comment by sytsereitsma** [comment]
Unit test is failing. The padding characters have changed from 00 to ==. Still seems to work, so please fix the unit test to use the new padding characters.

**Comment by danieleades** [comment]
> Unit test is failing. The padding characters have changed from 00 to ==. Still seems to work, so please fix the unit test to use the new padding characters.

ah rats. I think I only ran the unit tests in the `base64_plantuml` module....

should be working now

**Comment by sytsereitsma** [comment]
Hi Daniel. Can you resolve the conflicts after my merge of your other PR?

**Comment by danieleades** [comment]
rebased. this is ready for review

**Comment by sytsereitsma** [comment]
Working down the line... You'll have to do some more rebasing I'm afraid.

**Pull request merged**